test/test_e2e.py

```diff
@@ -57,17 +57,28 @@ def produce_message(message: str, channel: BlockingChannel) -> None:
     channel.basic_publish("watched-files", "", body=message.encode())
 
 
+def get_specification_from_file(instrument: str) -> Any:
+    """
+    Given an instrument, return the specification
+    :param instrument: The instrument for which specification to get
+    :return: The specification file contents
+    """
+    path = Path(f"test/test_data/specifications/{instrument.lower()}_specification.json")
+    if not path.exists():
+        path = Path(f"test_data/specifications/{instrument.lower()}_specification.json")
+    with path.open(encoding="utf-8") as fle:
```

Comment:
Though the only thing that stood out to me as a potential issue is the difference between these two paths. It's not obvious why they are different.

Reply:
Because we run tests in 2 contexts, locally, and on the CI, in the CI, it has a different working directory to the default selected by IDEs for local testing, largely this is just a small helper to ensure the data that is being loaded can be found to compare against, if this fails the test fails and it can be unclear as to why.

Reply:
Sure that makes sense. Perhaps that information could be included in the doc string? For any future new developer that might be working on this.